storm-core/src/jvm/backtype/storm/task/TopologyContext.java

```diff
@@ -205,32 +217,65 @@ public Object getTaskData(String name) {
     public void setExecutorData(String name, Object data) {
         _executorData.put(name, data);
     }
-    
+
     public Object getExecutorData(String name) {
         return _executorData.get(name);
-    }    
-    
+    }
+
     public void addTaskHook(ITaskHook hook) {
         hook.prepare(_stormConf, this);
         _hooks.add(hook);
     }
-    
+
     public Collection<ITaskHook> getHooks() {
         return _hooks;
     }
+
+    public Object groupingToJSONableObject(Grouping grouping) {
+    	if (grouping.is_set_fields()) {
+    		return grouping.get_fields();
+    	} else {
+    		return grouping.getSetField().toString();
+    	}
```

Comment:
I understand wanting to make something that's easily extensible in the future, but the example you showed wouldn't ever happen, because only the fields grouping specifies fields.  If you think it should return a dictionary that always contains `type` and `fields`, we'd get the following for shuffle groupings:

```json
{
    "type": "SHUFFLE",
    "fields": null
}
```

and 

```json
{
    "type": "FIELDS",
    "fields": ["a", "b"]
}
```

for fields groupings.